cmd/dfget/app/server.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/dragonflyoss/Dragonfly/dfget/core/uploader"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+	"github.com/spf13/cobra"
+)
+
+var serverCmd = &cobra.Command{
+	Use:   "server",
+	Short: "Launch a peer server for uploading files.",
+	Run: func(cmd *cobra.Command, args []string) {
+		logPath := filepath.Join(cfg.WorkHome, "logs")
+		cfg.ServerLogger = util.CreateLogger(logPath,
+			"dfserver.log", "INFO", cfg.Sign)
+		if port, err := uploader.LaunchPeerServer(cfg); err == nil {
+			fmt.Println(port)
+			uploader.WaitForShutdown()
+		} else {
+			fmt.Println(err.Error())
```

Comment:
I think we should exit with a non-0 exit code here, while `fmt.Println` will never return a non-zero exit code.

In addition, I wish that we could first check the case of `err != nil`. Because this could make the exception handling in an indent loop, but the main logic could remain in the main code block.